libcaf_net/caf/net/http/server_factory.cpp

```diff
@@ -175,8 +175,10 @@ expected<disposable> do_start_impl(Config& cfg, Acceptor acc,
                                             cfg.max_connections,
                                             cfg.monitored_actors);
   auto ptr = net::socket_manager::make(cfg.mpx, std::move(impl));
-  cfg.mpx->start(ptr);
-  return expected<disposable>{disposable{std::move(ptr)}};
+  if (cfg.mpx->start(ptr))
+    return expected<disposable>{disposable{std::move(ptr)}};
+  return make_error(sec::disposed,
```

Comment:
I don't think `disposed` is the right error code for this situation. Above, we use `logic_error` in a similar case. Applies to all instances where we return `disposed` for this kind of error.